### PR TITLE
added support for other exstension types

### PR DESCRIPTION
**Pull request by P3ntest**

### CODE REVIEW

File: pagan/pagan.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ def change(self, inpt, hashfun=DEFAULT_HASHFUN):
         Uses the standard hash function if no one is given."""
         self.img = self.__create_image(inpt, hashfun)
 
-    def save(self, path=DEFAULT_OUTPUT_PATH, filename=DEFAULT_FILENAME):
+    def save(self, path=DEFAULT_OUTPUT_PATH, filename=DEFAULT_FILENAME, exstension):
```
Comment (ofalk) [inline pagan/pagan.py:41]: typo. => extension

File: pagan/pagan.py [hunk 2]
```diff
@@ -50,12 +50,15 @@ def save(self, path=DEFAULT_OUTPUT_PATH, filename=DEFAULT_FILENAME):
             os.makedirs(path)
 
         # Cut the .png file ending if one was omitted.
-        if filename[-4:] == ".png":
+        if !str.startswith(extensionT, ".")
```
Comment (ofalk) [inline pagan/pagan.py:53]: where does this trailing uppercase T come frome? and it's not the same as the above exstension, which should have been extension.

File: pagan/pagan.py [hunk 3]
```diff
@@ -50,12 +50,15 @@ def save(self, path=DEFAULT_OUTPUT_PATH, filename=DEFAULT_FILENAME):
...
+        	extension = str.subtext(exstension, 1 - len(exstension));
```
Comment (ofalk) [inline pagan/pagan.py:54]: if you wanted to make a difference between this 'extension' here and the above - i'd go for final_extension or something like that.

File: pagan/pagan.py [hunk 4]
```diff
@@ -50,12 +50,15 @@ def save(self, path=DEFAULT_OUTPUT_PATH, filename=DEFAULT_FILENAME):
             os.makedirs(path)
 
         # Cut the .png file ending if one was omitted.
-        if filename[-4:] == ".png":
+        if !str.startswith(extensionT, ".")
+        	extension = str.subtext(exstension, 1 - len(exstension)); 
+
+        if filename[-4:] == "." + exstension:
```
Comment (ofalk) [inline pagan/pagan.py:56]: see above.

File: pagan/pagan.py [hunk 5]
```diff
@@ -50,12 +50,15 @@ def save(self, path=DEFAULT_OUTPUT_PATH, filename=DEFAULT_FILENAME):
...
             filename = filename[:-4]
 
         # Saves the image under the given filepath.
-        filepath = ("%s%s.png" % (path, filename))
-        filepath = os.path.join(path, "%s.png" % filename)
+        filepath = ("%s%s." + exstension % (path, filename))
```
Comment (ofalk) [inline pagan/pagan.py:60]: see above.

File: pagan/pagan.py [hunk 6]
```diff
@@ -50,12 +50,15 @@ def save(self, path=DEFAULT_OUTPUT_PATH, filename=DEFAULT_FILENAME):
...
+        filepath = os.path.join(path, "%s." + exstension % filename)
```
Comment (ofalk) [inline pagan/pagan.py:61]: see above.

File: pagan/pagan.py [hunk 7]
```diff
@@ -50,12 +50,15 @@ def save(self, path=DEFAULT_OUTPUT_PATH, filename=DEFAULT_FILENAME):
...
         # FIXIT: filepath without SUFFIX, print writes false filename
         print ("Saving: %s" % filepath)
-        self.img.save(filepath, 'PNG')
+        self.img.save(filepath, str.upper(exstension)
```
Comment (ofalk) [inline pagan/pagan.py:64]: see above.

**Review by ofalk — CHANGES_REQUESTED** [review]
It's not my call, but overall this code isn't clean and shouldn't be merged this way in my honest opinion.
However, I'd appreciate the support for different extensions.

@daboth I hope you don't mind I reviewed this.

**Comment by daboth** [comment]
Thank you @ofalk for reviewing. I also think it would be nice to include more extensions, but i need to do some testing first (e.g. picture quality, ect. for non-png images). I would also prefer to use constants for supported extensions instead of a simple variable.

**Comment by ofalk** [comment]
Sorry for the noise here. I've reworked the idea in pull request #11 but forgot to squash correctly. I'd suggest to close this pull in favour of mine :-)

**Comment by daboth** [comment]
Closed in favor of #11. I appreciate every commit though, the idea was good! Will test and merge #11 when i have some spare time :)

**Pull request closed without merging**